tf2_web_republisher/src/tf2_web_republisher.cpp

```diff
@@ -0,0 +1,138 @@
+/*********************************************************************
+ *
+ *  Copyright (c) 2014, Willow Garage, Inc.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the Willow Garage nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+
+ *  Author: Julius Kammerl (jkammerl@willowgarage.com)
+ *
+ */
+
+#include "sstream"
+
+#include "tf_web_republisher/tf2_web_republisher.hpp"
+
+#include "geometry_msgs/msg/transform_stamped.hpp"
+#include "rclcpp_action/rclcpp_action.hpp"
+#include "tf2/transform_datatypes.h"
+#include "tf2_ros/buffer.h"
+#include "tf2_web_republisher_msgs/action/tf_subscription.hpp"
+#include "tf2_web_republisher_msgs/srv/republish_t_fs.hpp"
+#include <tf2_ros/transform_listener.h>
+
+TFRepublisher::TFRepublisher(const std::string& name, const rclcpp::NodeOptions& options) : Node(name, options)
+{
+  using namespace std::placeholders;
+
+  action_server_ = rclcpp_action::create_server<tf2_web_republisher_msgs::action::TFSubscription>(
+      this, "tf2_web_republisher",
+      [this](auto&& PH1, auto&& PH2) {
+        return handle_goal(std::forward<decltype(PH1)>(PH1), std::forward<decltype(PH2)>(PH2));
+      },
+      [this](auto&& PH1) { return handle_cancel(std::forward<decltype(PH1)>(PH1)); },
+      [this](auto&& PH1) { handle_accepted(std::forward<decltype(PH1)>(PH1)); });
+
+  tf_buffer_ = std::make_unique<tf2_ros::Buffer>(this->get_clock());
```

Comment:
nit: you can probably just do `get_clock()` instead of `this->get_clock()`

Reply:
I tend to always add `this->` when calling class methods to differentiate them from functions. I think either way is good, as long as the calling convention is consistent across the project (class methods are always called with `this->` or always without it, eventually only private methods).